app/client/components/user.tsx

```diff
@@ -27,12 +25,58 @@ export interface MeResponse {
   };
 }
 
-export class MeCheckerAsyncLoader extends AsyncLoader<MeResponse> {}
+export interface Subscription {
+  start: string;
+  end: string;
+  cancelledAt: boolean;
+  nextPaymentDate: string;
+  card?: CardProps;
+  plan: {
+    amount: number;
+    currency: string;
+  };
+}
+
+export interface HasSubscription {
+  subscription: Subscription;
+}
+
+const zuoraCancellationReasonMapping: { [zuoraReasonKey: string]: string } = {
+  mma_editorial: "I am unhappy with Guardian journalism",
+  mma_support_another_way:
+    "I am going to support The Guardian in another way, eg. by subscribing",
+  mma_values: "I don't feel that the Guardian values my support",
+  mma_payment_issue: "I didn't expect The Guardian to take another payment",
+  mma_health: "Ill-health",
+  mma_none: "None of the membership benefits are of interest to me",
+  mma_financial_circumstances: "A change in my financial circumstances",
+  mma_other: "Other"
+};
 
 export const fetchMe: () => Promise<MeResponse> = async () => {
   return (await fetch("/api/me", { credentials: "include" })).json();
 };
 
+export const CancellationReasonContext: React.Context<
+  string
+> = React.createContext("");
+
+export const CancellationTypeContext: React.Context<
+  string
+> = React.createContext("error");
+
+export const CancellationUrlSuffixContext: React.Context<
```

Comment:
I'm not sure this is worth having in a context, the flow for friend tier membership should be separate. Also this `HasSubscriptionGetterContext` and `cancelType` seem to be managing the same piece of information.